webtau-cli/src/main/java/com/twosigma/webtau/cli/ProcessUtils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.cli;
+
+import java.io.IOException;
+
+public class ProcessUtils {
+    private ProcessUtils() {
+    }
+
+    public static ProcessRunResult run(String[] command) {
+        ProcessBuilder processBuilder = new ProcessBuilder(command);
+        try {
+            Process process = processBuilder.start();
+
+            StreamGobbler outputGobbler = new StreamGobbler("output", process.getInputStream());
+            StreamGobbler errorGobbler = new StreamGobbler("error", process.getErrorStream());
+
+            new Thread(errorGobbler).start();
+            new Thread(outputGobbler).start();
+
+            process.waitFor();
+
+            return new ProcessRunResult(process.exitValue(), outputGobbler.getLines(), errorGobbler.getLines());
```

Comment:
Could the reason for your sporadic test failures be that you're not joining the two gobbler threads?  They could in theory still be consuming the stream when you call `getLines` here.

Reply:
Trying

Reply:
You were right, thanks @tsiq-karold. Wonder if executor would have been better in this case.

Reply:
That thought had occurred to me.  However, it won't prevent you from having to join conceptually, instead you'll call get on a future.  You'll also have to work out how to size the executor or I suppose let the JVM do that for you and hope it's never an issue (which for this usecase it likely won't be).